hw3/3-5/src/App.js

```diff
@@ -0,0 +1,19 @@
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import Layout from "./pages/Layout";
+import Home from "./pages/Home";
+import Search from "./pages/Search";
+import Houses from "./pages/Houses";
+
+export default function App() {
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<Layout />}>
+          <Route index element={<Home />} />
+          <Route path="search" element={<Search />} />
+          <Route path="houses" element={<Houses />} />
+        </Route>
+      </Routes>
+    </BrowserRouter>
+  );
+}
```

Comment:
Fetching the data (calling the exact same API) twice, in both the Houses and Search components is inefficient. The data they both rely on is the same, and additionally both of these components have the same parent.

So why not let the data flow from parent to child?

It would make a lot more sense to call the API in the App component, then pass the data as props to the houses and search components. This would be more efficient since the API would only need to be called once, but the data would be available across the components that need it.

In summary, you can just fetch the data once and then pass it to child components via props.